### DISCUSSION TITLE
Replicate `git commit --only`

### DISCUSSION THREAD

**Discussion by chrisjsewell** [gitpython-developers/GitPython#1145] [Q&A] [answered]
Apologies if this is a 'newbie' question, but is there a way to commit only certain paths from the index tree, akin to `git commit --only file.txt`?

**Comment by Imipenem** [comment]
This would be very interesting to know, since I can not find anything like this in the docs or somewhere

**Comment by Byron** [comment, marked as answer]
I think if in doubt, and for these one-off commands, one can start off [using git directly](https://gitpython.readthedocs.io/en/stable/tutorial.html#using-git-directly). That way you can basically write simple scripts closely resembling git invocations in python.

Otherwise one would have to implement this functionality using the plumbing (low-level functionality) provided by GitPython, which isn't what anyone should start out with.

Even though I close this issue, please feel free posting follow-up questions here for eventual response.
